src/core/evaluator.js

```diff
@@ -1695,6 +1755,21 @@ class PartialEvaluator {
             // e.g. as done in https://github.com/mozilla/pdf.js/pull/6266,
             // but doing so is meaningless without knowing the semantics.
             continue;
+          case OPS.beginMarkedContentProps:
+            if (!isName(args[0])) {
+              warn(`Expected name for beginMarkedContentProps arg0=${args[0]}`);
+              continue;
+            }
+            fn = OPS.beginMarkedContentProps;
+            if (args[0].name === "OC") {
+              args = ["OC", self.parseMarkedContentProps(args[1], resources)];
```

Comment:
With `parseMarkedContentProps` being asynchronous, as suggested above, this would need to become something like this (which thus handles any errors gracefully):
```javascript
next(
  self.parseMarkedContentProps(args[1], resources)
    .then(data => {
       operatorList.addOp(OPS.beginMarkedContentProps, ["OC", data]);
    })
    .catch(reason => {
      if (reason instanceof AbortException) {
        return;
      }
      if (self.options.ignoreErrors) {
        self.handler.send("UnsupportedFeature", {
          featureId: /* Some new UNSUPPORTED_FEATURES id */,
        });
        warn(`getOperatorList - ignoring beginMarkedContentProps: "${reason}".`);
        return;
      }
      throw reason;
    })
);
return;
```